transformer_engine/paddle/csrc/custom_ops.cu

```diff
@@ -1277,6 +1344,58 @@ void amax_and_scale_update_inplace(paddle::Tensor &amax_history,  // NOLINT
       static_cast<NVTEDType>(fp8_dtype), margin, amax_history.stream());
 }
 
+void amax_and_scale_update_inplace_legacy(paddle::Tensor &amax_history,  // NOLINT
+                                          paddle::Tensor &scale,         // NOLINT
+                                          paddle::Tensor &scale_inv,     // NOLINT
+                                          const paddle::Tensor &non_weight_mask,
+                                          const paddle::Tensor &current_step_id_tensor,
+                                          bool update_weight_scale_inv, float fp8_max, float margin,
+                                          const std::string &amax_compute) {
+  NVTE_CHECK(amax_compute == "max" || amax_compute == "most_recent");
+
+  paddle::Tensor amax;
+
+  if (amax_compute == "max") {
+    amax = amax_history.max({0});
+  } else {
+    amax = amax_history.slice(0, 1);
+  }
+
+  const auto rolled_amax_history = amax_history.roll({-1}, {0});
+
+  auto amax_history_numel = amax_history.numel();
+  auto amax_numel = amax.numel();
+  size_t num_blocks = (amax_history_numel + BLOCK_SIZE - 1) / BLOCK_SIZE;
+
+  auto parameterSetter = [current_step_id_tensor](phi::backends::gpu::CUDAKernelParams &params) {
```

Comment:
I'm not sure how expensive it is to copy a Paddle tensor, but no harm capturing-by-reference to avoid triggering the copy-constructor:
```suggestion
  auto parameterSetter = [&current_step_id_tensor](phi::backends::gpu::CUDAKernelParams &params) {
```